compiler/rustc_session/src/session.rs

```diff
@@ -1419,14 +1419,17 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
         );
     }
 
-    // LLVM CFI requires LTO.
-    if sess.is_sanitizer_cfi_enabled() {
-        if sess.opts.cg.lto == config::LtoCli::Unspecified
-            || sess.opts.cg.lto == config::LtoCli::No
-            || sess.opts.cg.lto == config::LtoCli::Thin
-        {
+    // LLVM CFI and VFE both require LTO.
+    if sess.opts.cg.lto == config::LtoCli::Unspecified
```

Comment:
This, I feel, should be using `sess.lto() == config::Lto::Fat` so that e.g. a specific target implying fat LTO is also taken into account here.